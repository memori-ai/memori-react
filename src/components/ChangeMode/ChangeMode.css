.memori--changeMode-instruct {
    padding: 1rem;
    border-radius: 5px;
}

.memori--changeMode-instruct-radio {
    display: flex;
}

.memori--changeMode-instruct-radio-button:first-of-type button {
    border-bottom-right-radius: 0;
    border-top-right-radius: 0;
}

.memori--changeMode-instruct-radio-button:last-of-type button {
    border-bottom-left-radius: 0;
    border-top-left-radius: 0;
}

.memori--changeMode-instruct-radio-button[data-headlessui-state='checked'] button {
    font-weight: 600;
}
