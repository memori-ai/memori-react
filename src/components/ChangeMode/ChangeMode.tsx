import React from 'react';
import { RadioGroup } from '@headlessui/react';
import { useTranslation } from 'react-i18next';
import Button from '../ui/Button';

export interface Props {
  instruct?: boolean;
  onChangeMode: (mode: 'test' | 'instruct') => void;
  canInstruct?: boolean;
}

const ChangeMode: React.FC<Props> = ({
  instruct,
  canInstruct,
  onChangeMode,
}) => {
  const { t } = useTranslation();

  return canInstruct ? (
    <div className="memori--changeMode-instruct">
      <RadioGroup
        name="instruct"
        value={instruct ? 'instruct' : 'test'}
        defaultValue={instruct ? 'instruct' : 'test'}
        className="memori--changeMode-instruct-radio"
        onChange={onChangeMode}
      >
        <RadioGroup.Option
          value="instruct"
          className="memori--changeMode-instruct-radio-button"
        >
          {({ checked }) => (
            <Button primary={checked}>
              {t('widget.instruct') || 'Instruct'}
            </Button>
          )}
        </RadioGroup.Option>
        <RadioGroup.Option
          value="test"
          className="memori--changeMode-instruct-radio-button"
        >
          {({ checked }) => (
            <Button primary={checked}>{t('widget.test') || 'Test'}</Button>
          )}
        </RadioGroup.Option>
      </RadioGroup>
    </div>
  ) : null;
};

export default ChangeMode;
