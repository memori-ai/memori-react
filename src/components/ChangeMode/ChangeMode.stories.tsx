import React from 'react';
import { Meta, Story } from '@storybook/react';
import I18nWrapper from '../../I18nWrapper';
import ChangeMode, { Props } from './ChangeMode';

import './ChangeMode.css';

const meta: Meta = {
  title: 'Widget/Change Mode',
  component: ChangeMode,
  argTypes: {
    instruct: {
      control: {
        type: 'boolean',
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<Props> = args => {
  const [instruct, setInstruct] = React.useState(args.instruct);

  return (
    <I18nWrapper>
      <ChangeMode
        {...args}
        canInstruct
        instruct={instruct}
        onChangeMode={mode => setInstruct(mode === 'instruct')}
      />
    </I18nWrapper>
  );
};

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  instruct: false,
};

export const Instruct = Template.bind({});
Instruct.args = {
  instruct: true,
};
