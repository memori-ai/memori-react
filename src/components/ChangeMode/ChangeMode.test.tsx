import React from 'react';
import { render } from '@testing-library/react';
import ChangeMode from './ChangeMode';

it('renders ChangeMode in test mode unchanged', () => {
  const { container } = render(
    <ChangeMode instruct={false} canInstruct onChangeMode={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});

it('renders ChangeMode in instruct mode unchanged', () => {
  const { container } = render(
    <ChangeMode instruct={true} canInstruct onChangeMode={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});

it('renders ChangeMode null if cannot set instruct unchanged', () => {
  const { container } = render(
    <ChangeMode instruct={false} canInstruct={false} onChangeMode={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});
