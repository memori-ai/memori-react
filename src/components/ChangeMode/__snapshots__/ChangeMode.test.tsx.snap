// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChangeMode in instruct mode unchanged 1`] = `
<div>
  <div
    class="memori--changeMode-instruct"
  >
    <input
      checked=""
      hidden=""
      name="instruct"
      readonly=""
      style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
      type="radio"
      value="instruct"
    />
    <div
      class="memori--changeMode-instruct-radio"
      id="headlessui-radiogroup-:r3:"
      role="radiogroup"
    >
      <div
        aria-checked="true"
        class="memori--changeMode-instruct-radio-button"
        data-headlessui-state="checked"
        id="headlessui-radiogroup-option-:r4:"
        role="radio"
        tabindex="0"
      >
        <button
          class="memori-button memori-button--primary memori-button--rounded memori-button--padded"
        >
          widget.instruct
        </button>
      </div>
      <div
        aria-checked="false"
        class="memori--changeMode-instruct-radio-button"
        data-headlessui-state=""
        id="headlessui-radiogroup-option-:r5:"
        role="radio"
        tabindex="-1"
      >
        <button
          class="memori-button memori-button--rounded memori-button--padded"
        >
          widget.test
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ChangeMode in test mode unchanged 1`] = `
<div>
  <div
    class="memori--changeMode-instruct"
  >
    <input
      checked=""
      hidden=""
      name="instruct"
      readonly=""
      style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
      type="radio"
      value="test"
    />
    <div
      class="memori--changeMode-instruct-radio"
      id="headlessui-radiogroup-:r0:"
      role="radiogroup"
    >
      <div
        aria-checked="false"
        class="memori--changeMode-instruct-radio-button"
        data-headlessui-state=""
        id="headlessui-radiogroup-option-:r1:"
        role="radio"
        tabindex="-1"
      >
        <button
          class="memori-button memori-button--rounded memori-button--padded"
        >
          widget.instruct
        </button>
      </div>
      <div
        aria-checked="true"
        class="memori--changeMode-instruct-radio-button"
        data-headlessui-state="checked"
        id="headlessui-radiogroup-option-:r2:"
        role="radio"
        tabindex="0"
      >
        <button
          class="memori-button memori-button--primary memori-button--rounded memori-button--padded"
        >
          widget.test
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ChangeMode null if cannot set instruct unchanged 1`] = `<div />`;
